integr8ly/tutorial-web-app-operator
# INTLY-2346 add openshift 4 support

## OpenShift 3 Verification
- Have an OpenShift 3 cluster with Integreatly installed
- Update the webapp operator image to point to `docker.io/aidenkeatingrht/tutorial-web-app-operator:INTLY-2346-openshift-4-support`
- Delete the webapp deploymentconfig
- Ensure the created webapp deploymentconfig has it's OPENSHIFT_VERSION env  set as `3`
- Go through Walkthrough 1A and ensure it still works

## OpenShift 4 Verification
- Have an OpenShift 4 cluster with Integreatly installed
- Run `oc new-project intly-2346-verification`
- Run `make cluster/prepare`
- Run `sed -i -e 's/quay.io\/integreatly\/tutorial-web-app-operator:v0.0.27/docker.io\/aidenkeatingrht\/tutorial-web-app-operator:INTLY-2346-openshift-4/g' deploy/operator.yaml`
- Ensure any existing `WebApp` resources are removed
- Run `make cluster/deploy`
- Create or modify an existing `WebApp` resource, similar to the following, in the namespace
```
apiVersion: integreatly.org/v1alpha1
kind: WebApp
metadata:
  name: solution-explorer
spec:
  app_label: tutorial-web-app
  template:
    parameters:
      OPENSHIFT_HOST: https://console-openshift-console.apps.cluster-rhmi-c41a.rhmi-c41a.openshiftworkshop.com
      OPENSHIFT_OAUTH_HOST: oauth-openshift.apps.cluster-rhmi-c41a.rhmi-c41a.openshiftworkshop.com
      OPENSHIFT_OAUTHCLIENT_ID: integreatly-solution-explorer
      SSO_ROUTE: https://sso-integreatly-rhsso.apps.cluster-rhmi-c41a.rhmi-c41a.openshiftworkshop.com
      OPENSHIFT_VERSION: '4'
    path: /home/tutorial-web-app-operator/deploy/template/tutorial-web-app.yml
```
- Run `oc edit oauthclient integreatly-solution-explorer` and add the webapp URL to the redirect URI's
- Open the webapp URI, go through Walkthrough 1A and ensure
  - Fuse can be opened
  -  AMQ Online can be opened
  - AMQ Online credentials are provided
